docs/getting-started/quickstart.rst

```diff
@@ -338,6 +331,14 @@ When we pass in ``"bhavik@example.com"`` as
 the actor, the rule no longer succeeds because the string ``"bhavik@example.com"`` does not
 match the string ``"alice@example.com"``.
 
+A Quick Note on Type Checking
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+You may have already guessed that the ``Expense`` term following the colon in the head of our policy rule
+is specifying an argument type. This is a :ref:`specializer <Specialization>` that controls rule
+execution based on whether it matches the supplied argument. In general, arguments match specializers
+if they are an instance of the specializer class. Here, we specialize the third argument on
+our own ``Expense`` class. We'll see more examples of specializers later in this guide.
+
```

Comment:
Maybe add something like:

"Specializers are optional but highly recommended in order to avoid potential policy bugs when an unexpected type of resource is passed to a certain rule".

(But worded better?)